### PR TITLE
Added basic support for SAMR

### PR DESCRIPTION
**Pull request by bli-r7**
Changes:
- Added connect2 support
- Added openDomain support
- Refactor ContextHandle to support variable length

### CODE REVIEW

File: src/main/java/com/rapid7/client/dcerpc/mssamr/messages/SamrConnect2Request.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        throws IOException {
+        packetOut.writeStringRef(serverName, true);
+        packetOut.writeInt((int) EnumUtils.toLong(desiredAccess));
+    }
+
+    @Override
+    public SamrConnect2Response getResponseObject() {
+        return new SamrConnect2Response();
+    }
+    /////////////////////////////////////////////////////////////////////////
```
Comment (sangdo-lee) [inline src/main/java/com/rapid7/client/dcerpc/mssamr/messages/SamrConnect2Request.java]: minor, if dont' need the below empty comments, could you remove it ? :)

Comment (bli-r7) [reply src/main/java/com/rapid7/client/dcerpc/mssamr/messages/SamrConnect2Request.java]: yes.

File: src/main/java/com/rapid7/client/dcerpc/messages/HandleResponse.java [hunk 2]
```diff
@@ -59,17 +59,25 @@ public int getReturnValue() {
...
-        handle.setBytes(handleBytes);
+        // Operation: OpenHKLM (2)
+        // [Request in frame: 11174]
+        // Pointer to Handle (policy_handle)
+        // Policy Handle: OpenHKLM(<...>)
+        // Handle: 0000000032daf234b77c86409d29efe60d326683
+        // [Frame handle opened: 11176]
+        // [Frame handle closed: 11424]
+        // Windows Error: WERR_OK (0x00000000)
+	handle.unmarshall(packetIn);
```
Comment (tosmun-r7) [inline src/main/java/com/rapid7/client/dcerpc/messages/HandleResponse.java]: Indentation

File: src/main/java/com/rapid7/client/dcerpc/objects/ContextHandle.java [hunk 3]
```diff
@@ -74,4 +87,12 @@ public int hashCode() {
     public boolean equals(final Object anObject) {
         return anObject instanceof ContextHandle && Arrays.equals(handle, ((ContextHandle) anObject).handle);
     }
+
+    @Override
+    public ContextHandle unmarshall(PacketInput in) throws IOException {
```
Comment (tosmun-r7) [inline src/main/java/com/rapid7/client/dcerpc/objects/ContextHandle.java]: :+1:

File: src/main/java/com/rapid7/client/dcerpc/mslsad/messages/LsarOpenPolicy2Request.java [hunk 4]
```diff
@@ -23,8 +23,8 @@
 import com.hierynomus.msdtyp.AccessMask;
 import com.hierynomus.protocol.commons.EnumWithValue.EnumUtils;
 import com.rapid7.client.dcerpc.io.PacketOutput;
+import com.rapid7.client.dcerpc.messages.HandleResponse;
 import com.rapid7.client.dcerpc.messages.RequestCall;
-import com.rapid7.client.dcerpc.msrrp.messages.HandleResponse;
 
 public class LsarOpenPolicy2Request extends RequestCall<HandleResponse> {
```
Comment (sangdo-lee) [inline src/main/java/com/rapid7/client/dcerpc/mslsad/messages/LsarOpenPolicy2Request.java]: it it correct?

Comment (bli-r7) [reply src/main/java/com/rapid7/client/dcerpc/mslsad/messages/LsarOpenPolicy2Request.java]: The package was moved.

File: src/main/java/com/rapid7/client/dcerpc/mssrvs/ServerService.java [hunk 5]
```diff
@@ -35,11 +35,15 @@
 import com.rapid7.client.dcerpc.transport.RPCTransport;
 
 /**
- * This class implements a partial service service in accordance with [MS-SRVS]: Server Service Remote Protocol which
- * specifies the Server Service Remote Protocol, which remotely enables file and printer sharing and named pipe access
+ * This class implements a partial service service in accordance with [MS-SRVS]:
```
Comment (bli-r7) [inline src/main/java/com/rapid7/client/dcerpc/mssrvs/ServerService.java]: I'll revert the changes to this.

**Pull request merged**